### PR TITLE
Home task2

### PR DESCRIPTION
**Pull request by elenagronskaya**

### CODE REVIEW

File: src/components/Registration/Registration.jsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+			// setRegistration({ name: '', email: '', password: '' });
+			try {
+				await doRegistration(
+					registration.name,
+					registration.email,
+					registration.password
+				);
+				navigate('/login');
+			} catch (error) {
+				debugger;
```
Comment (urakovaliaskar) [inline src/components/Registration/Registration.jsx]: remove

File: src/components/CourseInfo/CourseInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	useEffect(() => {
+		setCourse(getCourseById(courseId));
+	}, [courseId]);
+
+	const createDate = moment().format('DD.MM.YYYY');
+
+	return (
+		<section className={styles.courseInfo}>
+			<Link to='/courses' className={styles.link}>
```
Comment (urakovaliaskar) [inline src/components/CourseInfo/CourseInfo.jsx]: move routes to const and reuse them

File: src/components/CourseInfo/CourseInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+				&lt;Back to courses
```
Comment (urakovaliaskar) [inline src/components/CourseInfo/CourseInfo.jsx]: Add space

File: src/components/Registration/Registration.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	const handleChange = (e) => {
+		const name = e.target.name;
+		const value = e.target.value;
+		setRegistration({ ...registration, [name]: value });
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		if (registration.name && registration.email && registration.password) {
+			// setRegistration({ name: '', email: '', password: '' });
```
Comment (urakovaliaskar) [inline src/components/Registration/Registration.jsx]: What is this for? remove.

File: src/common/Button/Button.jsx [hunk 5]
```diff
@@ -1,13 +1,22 @@
...
 		className={showCourseButtonStyle ? showCourseButtonStyle : styles.button}
 		onClick={onClick}
+		type={type}
 	>
 		{buttonText}
 	</button>
 );
 
+Button.propTypes = {
+	buttonText: PropTypes.string,
```
Comment (urakovaliaskar) [inline src/common/Button/Button.jsx]: buttonText is used in every Button those it should be required prop. `PropTypes.string.isRequired`
For not required props define default value, for example:
`Button.defaultProps = {
    type: 'button'
}`

Comment (urakovaliaskar) [reply src/common/Button/Button.jsx]: Do it for all components with proptypes

File: src/components/CreateCourse/CreateCourse.jsx [hunk 6]
```diff
@@ -28,6 +31,12 @@ const CreateCourse = ({ setIsAddingCourse }) => {
 			Number(duration) > 0
 		);
 	};
+	useEffect(() => {
```
Comment (urakovaliaskar) [inline src/components/CreateCourse/CreateCourse.jsx]: add empty line, useEffect usually placed at the bottom of the functions

File: src/components/CreateCourse/CreateCourse.jsx [hunk 7]
```diff
@@ -28,6 +31,12 @@ const CreateCourse = ({ setIsAddingCourse }) => {
...
+		if (!isLoggedIn()) {
+			navigate('/login');
+			return;
```
Comment (urakovaliaskar) [inline src/components/CreateCourse/CreateCourse.jsx]: why do we need it? remove

File: src/components/CreateCourse/components/AuthorItem/AuthorItem.jsx [hunk 8]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import Button from '../../../../common/Button/Button';
+import PropTypes from 'prop-types';
```
Comment (urakovaliaskar) [inline src/components/CreateCourse/components/AuthorItem/AuthorItem.jsx:2]: to the top and add space to make a distinction between libraries and custom imports for all components

File: src/components/Header/Header.jsx [hunk 9]
```diff
@@ -1,17 +1,36 @@
...
+		localStorage.removeItem(USER_NAME);
+		setUserName('');
+		navigate('/login');
+	};
+	return (
+		<section className={styles.wrapperHeader}>
+			<Logo />
+			<div className={styles.wrapperButton}>
+				<span className={styles.nameAccount}>{userName}</span>
+				{userName && userName !== '' && (
```
Comment (urakovaliaskar) [inline src/components/Header/Header.jsx]: if userName is truthy, then it cannot be "" (empty string), remove userName !== ''

File: src/components/Registration/Registration.jsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+			/>
+			<Button buttonText={REGISTRATION} type='submit' />
+			<p className={styles.text}>
+				If you have an account you can
+				<Link to='/login' className={styles.link}>
+					{' '}
+					Login
+				</Link>
+			</p>
+			{errors.map((error) => {
```
Comment (urakovaliaskar) [inline src/components/Registration/Registration.jsx]: simplify, remove unnecessary return

File: src/services/authService.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios';
+
+export const doRegistration = async (name, email, password) => {
+	let payload = { name, email, password };
+	return await axios.post('http://localhost:4000/register', payload);
```
Comment (urakovaliaskar) [inline src/services/authService.js]: create const BASE_URL and reuse

File: src/services/courseService.js [hunk 12]
```diff
@@ -16,6 +16,21 @@ export const getCourses = (filter) => {
 		});
 };
 
+export const getCourseById = (id) => {
+	const courses = mockedCoursesList
+		.filter((item) => item.id.toLowerCase().includes(id.toLowerCase()))
+		.map((item) => {
+			const author = { ...item };
```
Comment (urakovaliaskar) [inline src/services/courseService.js]: Semantically speaking, it should be called course, not author

File: src/services/courseService.js [hunk 13]
```diff
@@ -16,6 +16,21 @@ export const getCourses = (filter) => {
...
+			author.authors = joinAuthors(item.authors);
+			return author;
+		});
+	if (courses == null) return null;
+	if (courses.length === undefined) {
```
Comment (urakovaliaskar) [inline src/services/courseService.js]: In which cases can this 2 conditions happen?

File: src/services/courseService.js [hunk 14]
```diff
@@ -16,6 +16,21 @@ export const getCourses = (filter) => {
 		});
 };
 
+export const getCourseById = (id) => {
+	const courses = mockedCoursesList
+		.filter((item) => item.id.toLowerCase().includes(id.toLowerCase()))
```
Comment (urakovaliaskar) [inline src/services/courseService.js]: use find, item.id.toLowerCase() === id.toLowerCase()

File: src/components/CourseInfo/CourseInfo.jsx [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+			</Link>
+			<div>
+				<h3 className={styles.title}>{course == null ? '' : course.title}</h3>
+				<div className={styles.content}>
+					<div className={styles.description}>
+						{course == null ? '' : course.description}
```
Comment (urakovaliaskar) [inline src/components/CourseInfo/CourseInfo.jsx]: course?.description || ''

Comment (urakovaliaskar) [reply src/components/CourseInfo/CourseInfo.jsx]: same for all

**Review by urakovaliaskar — CHANGES_REQUESTED** [review]
waiting for fixes

File: src/components/CreateCourse/components/AuthorItem/AuthorItem.jsx [hunk 16]
```diff
@@ -13,4 +15,16 @@ const AuthorItem = ({ name, id, processAuthor, isDelete }) => (
...
 
+AuthorItem.propTypes = {
+	name: PropTypes.string.isRequired,
+	isDelete: PropTypes.bool,
+	processAuthor: PropTypes.func.isRequired,
+	id: PropTypes.string.isRequired,
+};
+AuthorItem.defaultProps = {
+	type: 'button',
+	showCourseButtonStyle: styles.button,
```
Comment (AliaskarUrakovTR) [inline src/components/CreateCourse/components/AuthorItem/AuthorItem.jsx]: This is not AuthorItem's prop

File: src/components/Login/Login.jsx [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+		email: '',
+		password: '',
+	});
+
+	let navigate = useNavigate();
+
+	useEffect(() => {
+		if (isLoggedIn()) {
+			navigate(ROUTE_COURSES);
+			return;
```
Comment (AliaskarUrakovTR) [inline src/components/Login/Login.jsx]: Why it is needed here?

File: src/services/courseService.js [hunk 18]
```diff
@@ -16,6 +16,21 @@ export const getCourses = (filter) => {
 		});
 };
 
+export const getCourseById = (id) => {
+	const foundCourse = mockedCoursesList.find(
+		(item) => item.id.toLowerCase() === id.toLowerCase()
+	);
+
+	if (foundCourse == null) {
```
Comment (AliaskarUrakovTR) [inline src/services/courseService.js]: if (!foundCourse)

**Review by AliaskarUrakovTR — APPROVED** [review]

**Review by urakovaliaskar — APPROVED** [review]

**Pull request merged**